src/components/Component.js

```diff
@@ -0,0 +1,29 @@
+class Component {
+  $target;
+  state;
+
+  constructor($target, props) {
+    this.$target = $target;
+    this.props = props;
+    this.setup();
+    this.render();
+  }
+  setup() {}
+  template() {
+    return "";
+  }
+
+  render() {
+    this.$target.innerHTML = this.template();
+    this.setEvent();
+  }
+
+  setEvent() {}
+
+  setState(newState) {
+    this.state = { ...this.setState, ...newState };
```

Comment:
이렇게 맞을까요?
```suggestion
    this.state = { ...this.state, ...newState };
```

그런데 에러가 나지 않아서 보니까, 사용하는 곳이 없었습니다..!
쓰이진 않지만, 일단 만들어두신 이유가 궁금합니다~!

Reply:
https://github.com/woowacourse/javascript-lunch/pull/195/commits/1d02e457e1959999d540970337c61dbc6602dda3

네 맞습니다.. 😭 처음 설계 당시에는 미래에 상태를 업데이트할 필요가 있을 것을 고려하여 했던 설계였는데 결국엔 상태 업데이트를 사용하지 않았습니다... 현재는 필요하지 않은 코드이기 때문에 제거했습니다!